### ISSUE TITLE
Error: "." is not in the SourceMap webpack-sources@1.4.0 

### ISSUE THREAD

**Issue by yongnanzhu** [webpack/webpack-sources#55]
Hi, we found that webpack-sources@1.4.0 cause this error, however webpack-sources@1.3.0 is ok.

the detail errors is listed bellow.
 
/cloud/source_code/node_modules/_source-map@0.6.1@source-map/lib/source-map-consumer.js:753
throw new Error('"' + relativeSource + '" is not in the SourceMap.');
^
Error: "." is not in the SourceMap.
at BasicSourceMapConsumer.SourceMapConsumer_sourceContentFor [as sourceContentFor] (/cloud/source_code/node_modules/_source-map@0.6.1@source-map/lib/source-map-consumer.js:753:13)
at /cloud/source_code/node_modules/_webpack-sources@1.4.0@webpack-sources/lib/applySourceMap.js:88:47

**Comment by evilebottnawi** [comment]
@yongnanzhu can you create minimum reproducible test repo?

**Comment by rut2** [comment]
Same issue for me as well in 1.4.0 

```
/gitlab-runner/node_modules/webpack-sources/lib/applySourceMap.js:15
	removeGeneratedCodeForSourceFile = false
	                                 ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/gitlab-runner/node_modules/webpack-sources/lib/SourceMapSource.js:13:22)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
```

**Comment by evilebottnawi** [comment]
Please create minimum reproducible test repo

**Comment by Aure77** [comment]
Same issue with node 5.4.1 & webpack@3.3.0 (with webpack-sources resolve at 1.4.0)

**Comment by rut2** [comment]
Isn't this is the problem, 
it is working windows machine but not on Ubuntu machine. How can I use @1.3.0 until it is resolved?  
![image](https://user-images.githubusercontent.com/4327606/62214627-11caaa80-b3c3-11e9-84a4-dc8471a6920c.png)

**Comment by evilebottnawi** [comment]
webpack version and node version?

**Comment by sokra** [comment]
Thanks for your reports. Please upgrade to 1.4.1 and open new issues if there are still issues.

**Comment by rut2** [comment]
@sokra Working like a charm 👍 
Great ! Thanks.

**Comment by dlong500** [comment]
@sokra It seems that this issue has reappeared, and when trying to track down the relevant code I'm finding that one of the two calls to `sourceContentFor()` in `applySourceMap.js` is no longer passing the second parameter that was added in 4c036a82673603932c9ede479073a89f6d217b79 to "avoid throwing when source is not in SourceMap".

The call to `sourceContentFor()` (on line 144 of the latest version of the file) is no longer passing a second parameter.  It seems there is something odd with commit 7ba205121c91bda479316ac2aa669f59e289fead (the very next one after 4c036a82673603932c9ede479073a89f6d217b79) that reverts the change in the function call.  Because the `nullOnMissing` parameter isn't true that causes an error when compiling my project after updating to terser-webpack-plugin 4.2.x (which started using webpack.sources for webpack@5).  Can someone look into this?  Thanks.

References:
https://github.com/webpack/webpack/issues/11469

**Comment by evilebottnawi** [comment]
@dlong500 Can you open new issue with examples or link on broken code?

**Comment by dlong500** [comment]
@evilebottnawi I'm having difficulty coming up with a simple example as I'm not sure why the `sourceContentFor()` function is only sometimes called with a period for the source parameter.

The main reason I was posting here is that the fix introduced in response to this issue was partially reverted, and I believe that reversion was inadvertent.  I can't find any commit message mentioning a reason for removing the second parameter and it looks like the reversion could have been due to a merge commit from a different branch.

The bottom line is that the code used in webpack-sources v1.4.3 doesn't cause an error when called via terser-webpack-plugin, but since terser-webpack-plugin started using webpack's version of webpack-sources (v2.x) the error occurs.  I was hoping someone with more expertise in the source map related code would be able to figure out what's going on.

**Comment by evilebottnawi** [comment]
@dlong500 Please:
- open new issue and describe it, even you can create reproducible test repo
- can you provide error screenshots/video/small example of code?

**Comment by chaffeqa** [comment]
I'm getting the same thing using terser-webpack-plugin v4.2.1, looks like:
```
92% sealing asset processing SourceMapDevToolPlugin server.js generate SourceMap(node:8236) UnhandledPromiseRejectionWarning: Error: "." is not in the SourceMap.
    at BasicSourceMapConsumer.SourceMapConsumer_sourceContentFor [as sourceContentFor] (proj/node_modules/source-map/lib/source-map-consumer.js:753:13)
    at proj/node_modules/webpack-sources/lib/applySourceMap.js:144:46
    at SourceNode_walk [as walk] (proj/node_modules/source-map/lib/source-node.js:230:9)
    at SourceNode_walk [as walk] (proj/node_modules/source-map/lib/source-node.js:226:13)
    at applySourceMap (proj/node_modules/webpack-sources/lib/applySourceMap.js:58:13)
    at SourceMapSource.node (proj/node_modules/webpack-sources/lib/SourceMapSource.js:198:11)
    at exports.getSourceAndMap (proj/node_modules/webpack-sources/lib/helpers.js:17:21)
    at SourceMapSource.sourceAndMap (proj/node_modules/webpack-sources/lib/SourceMapSource.js:184:10)
    at getTaskForFile (proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:77:30)
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:264:22
(node:8236) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

i was NOT getting that error using v4.1.0, but that was running into issues with OOM due to caching or something like that... anyways I can reproduce this every time on our codebase... i can access the failing asset if that helps solve it:

```ts
SourceMapSource {
  _valueAsString: `....`,
  _valueAsBuffer: undefined,
  _name: 'static/js/bundle.client.c838107f.js',
  _hasSourceMap: true,
  _sourceMapAsObject: {
    version: 3,
    sources: [ 'static/js/bundle.client.c838107f.js' ],
    names: [
      'self',
      'push',
      '81511',
      '__unused_webpack_module',
      '__unused_webpack_exports',
      '__webpack_require__',
      'Promise',
      'enable',
      'window',
      // ...
}}
```

**Comment by chaffeqa** [comment]
@evilebottnawi i figured something out... by litterally commenting out code until it worked, i was able to narrow down what was causing that error to the following:

* Any `enum` declaration that gets compiled

Problem is: we are using babel-typescript, so I dont know what that transpiles down to exactly... im going to try to find the output source to see what it would look like.  But i guess this is a starting spot

PS: I'm testing a webpack build for target of nodeJS 14.11.0

**Comment by chaffeqa** [comment]
```92% sealing asset processing SourceMapDevToolPlugin sql.js generate SourceMapError: "." is not in the SourceMap.
    at BasicSourceMapConsumer.SourceMapConsumer_sourceContentFor [as sourceContentFor] (proj/node_modules/source-map/lib/source-map-consumer.js:753:13)
    at proj/node_modules/webpack-sources/lib/applySourceMap.js:144:46
    at SourceNode_walk [as walk] (proj/node_modules/source-map/lib/source-node.js:230:9)
    at SourceNode_walk [as walk] (proj/node_modules/source-map/lib/source-node.js:226:13)
    at applySourceMap (proj/node_modules/webpack-sources/lib/applySourceMap.js:58:13)
    at SourceMapSource.node (proj/node_modules/webpack-sources/lib/SourceMapSource.js:198:11)
    at exports.getSourceAndMap (proj/node_modules/webpack-sources/lib/helpers.js:17:21)
    at SourceMapSource.sourceAndMap (proj/node_modules/webpack-sources/lib/SourceMapSource.js:184:10)
    at getTaskForFile (proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:77:30)
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:265:32
SourceMapSource {
  _valueAsString: 'var e={436:()=>{("undefined"!=typeof window?window:"undefined"!=typeof global?global:"undefined"!=typeof self?self:{}).SENTRY_RELEASE={id:"dc12c7c7c657ba2e6ef299ac14fa6f8360a3cd82"}},415:()=>{var e;!function(e){e.UNAVAILABLE="UNAVAILABLE",e.AVAILABLE="AVAILABLE"}(e||(e={})),console.log("Tried running: but failed with error: ".concat(e.AVAILABLE))}},n={};function o(r){if(n[r])return n[r].exports;var A=n[r]={exports:{}};return e[r](A,A.exports,o),A.exports}return o(436),o(415);',
  _valueAsBuffer: undefined,
  _name: 'sql.js',
  _hasSourceMap: true,
  _sourceMapAsObject: {
    version: 3,
    sources: [ 'sql.js' ],
    names: [
      '__webpack_modules__',
      '436',
      'window',
      'global',
      'self',
      'SENTRY_RELEASE',
      'id',
      '415',
      'AvailableOptionEnum',
      'console',
      'log',
      'concat',
      'AVAILABLE',
      '__webpack_module_cache__',
      '__webpack_require__',
      'moduleId',
      'exports',
      'module'
    ],
    mappings: 'AAAS,IAAIA,EAAsB,CAE7BC,IACA,MAEa,oBAAXC,OAAyBA,OAA2B,oBAAXC,OAAyBA,OAAyB,oBAATC,KAAuBA,KAAO,IAAIC,eAAe,CAACC,GAAG,6CAIzIC,IACA,KAEN,IAAIC,GAEJ,SAAWA,GACTA,EAAiC,YAAI,cACrCA,EAA+B,UAAI,YAFrC,CAGGA,IAAwBA,EAAsB,KAEjDC,QAAQC,IAAI,yCAAyCC,OAAOH,EAAoBI,cAOnEC,EAA2B,GAG/B,SAASC,EAAoBC,GAE5B,GAAGF,EAAyBE,GAC3B,OAAOF,EAAyBE,GAAUC,QAG3C,IAAIC,EAASJ,EAAyBE,GAAY,CAGjDC,QAAS,IAOV,OAHAhB,EAAoBe,GAAUE,EAAQA,EAAOD,QAASF,GAG/CG,EAAOD,QAQf,OADAF,EAAoB,KACbA,EAAoB'
  },
  _sourceMapAsString: undefined,
  _sourceMapAsBuffer: undefined,
  _hasOriginalSource: true,
  _originalSourceAsString: '/******/ var __webpack_modules__ = ({\n' +
    '\n' +
    '/***/ 436:\n' +
    '/***/ (() => {\n' +
    '\n' +
    `(typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {}).SENTRY_RELEASE={id:"dc12c7c7c657ba2e6ef299ac14fa6f8360a3cd82"};\n` +
    '\n' +
    '/***/ }),\n' +
    '\n' +
    '/***/ 415:\n' +
    '/***/ (() => {\n' +
    '\n' +
    'var AvailableOptionEnum;\n' +
    '\n' +
    '(function (AvailableOptionEnum) {\n' +
    '  AvailableOptionEnum["UNAVAILABLE"] = "UNAVAILABLE";\n' +
    '  AvailableOptionEnum["AVAILABLE"] = "AVAILABLE";\n' +
    '})(AvailableOptionEnum || (AvailableOptionEnum = {}));\n' +
    '\n' +
    'console.log("Tried running: but failed with error: ".concat(AvailableOptionEnum.AVAILABLE));\n' +
    '\n' +
    '/***/ })\n' +
    '\n' +
    '/******/ });\n' +
    '/************************************************************************/\n' +
    '/******/ // The module cache\n' +
    '/******/ var __webpack_module_cache__ = {};\n' +
    '/******/ \n' +
    '/******/ // The require function\n' +
    '/******/ function __webpack_require__(moduleId) {\n' +
    '/******/ \t// Check if module is in cache\n' +
    '/******/ \tif(__webpack_module_cache__[moduleId]) {\n' +
    '/******/ \t\treturn __webpack_module_cache__[moduleId].exports;\n' +
    '/******/ \t}\n' +
    '/******/ \t// Create a new module (and put it into the cache)\n' +
    '/******/ \tvar module = __webpack_module_cache__[moduleId] = {\n' +
    '/******/ \t\t// no module.id needed\n' +
    '/******/ \t\t// no module.loaded needed\n' +
    '/******/ \t\texports: {}\n' +
    '/******/ \t};\n' +
    '/******/ \n' +
    '/******/ \t// Execute the module function\n' +
    '/******/ \t__webpack_modules__[moduleId](module, module.exports, __webpack_require__);\n' +
    '/******/ \n' +
    '/******/ \t// Return the exports of the module\n' +
    '/******/ \treturn module.exports;\n' +
    '/******/ }\n' +
    '/******/ \n' +
    '/************************************************************************/\n' +
    '/******/ // module exports must be returned from runtime so entry inlining is disabled\n' +
    '/******/ // startup\n' +
    '/******/ // Load entry module and return exports\n' +
    '/******/ __webpack_require__(436);\n' +
    '/******/ return __webpack_require__(415);\n',
  _originalSourceAsBuffer: undefined,
  _hasInnerSourceMap: true,
  _innerSourceMapAsObject: {
    version: 3,
    sources: [
      'webpack://./node_modules/@sentry/webpack-plugin/src/sentry.loader.js??ruleSet[1].rules[7].use[0]!./node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js',
      'webpack://./src/scripts/sql.ts',
      'webpack/bootstrap',
      'webpack/startup'
    ],
    names: [ 'AvailableOptionEnum', 'console', 'log', 'AVAILABLE' ],
    mappings: ';;;;;AAAA,0HAA0H,kBAAkB,+C;;;;;;;ICAvIA,mB;;WAAAA,mB;AAAAA,qB;AAAAA,qB;GAAAA,mB,KAAAA,mB;;AAILC,OAAO,CAACC,GAAR,iDAAqDF,mBAAmB,CAACG,SAAzE,G;;;;;;SCJA;SACA;;SAEA;SACA;SACA;SACA;SACA;SACA;SACA;SACA;SACA;SACA;SACA;SACA;;SAEA;SACA;;SAEA;SACA;SACA;;;SCrBA;SACA;SACA;SACA;SACA',
    file: 'x',
    sourcesContent: [
      `(typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {}).SENTRY_RELEASE={id:"dc12c7c7c657ba2e6ef299ac14fa6f8360a3cd82"};`,
      'enum AvailableOptionEnum {\n' +
        '  UNAVAILABLE = "UNAVAILABLE",\n' +
        '  AVAILABLE = "AVAILABLE",\n' +
        '}\n' +
        'console.log(`Tried running: but failed with error: ${AvailableOptionEnum.AVAILABLE}`)\n',
      '// The module cache\n' +
        'var __webpack_module_cache__ = {};\n' +
        '\n' +
        '// The require function\n' +
        'function __webpack_require__(moduleId) {\n' +
        '\t// Check if module is in cache\n' +
        '\tif(__webpack_module_cache__[moduleId]) {\n' +
        '\t\treturn __webpack_module_cache__[moduleId].exports;\n' +
        '\t}\n' +
        '\t// Create a new module (and put it into the cache)\n' +
        '\tvar module = __webpack_module_cache__[moduleId] = {\n' +
        '\t\t// no module.id needed\n' +
        '\t\t// no module.loaded needed\n' +
        '\t\texports: {}\n' +
        '\t};\n' +
        '\n' +
        '\t// Execute the module function\n' +
        '\t__webpack_modules__[moduleId](module, module.exports, __webpack_require__);\n' +
        '\n' +
        '\t// Return the exports of the module\n' +
        '\treturn module.exports;\n' +
        '}\n' +
        '\n',
      '// module exports must be returned from runtime so entry inlining is disabled\n' +
        '// startup\n' +
        '// Load entry module and return exports\n' +
        '__webpack_require__(436);\n' +
        'return __webpack_require__(415);\n'
    ]
  },
  _innerSourceMapAsString: undefined,
  _innerSourceMapAsBuffer: undefined,
  _removeOriginalSource: true
}
Object.keys(asset.info): minimized
Object.keys(asset.source): _valueAsString,_valueAsBuffer,_name,_hasSourceMap,_sourceMapAsObject,_sourceMapAsString,_sourceMapAsBuffer,_hasOriginalSource,_originalSourceAsString,_originalSourceAsBuffer,_hasInnerSourceMap,_innerSourceMapAsObject,_innerSourceMapAsString,_innerSourceMapAsBuffer,_removeOriginalSource
Object.keys(asset.source._sourceMapAsObject): version,sources,names,mappings
Object.keys(asset.source._sourceMapAsObject.names): __webpack_modules__, 436, window, global, self, SENTRY_RELEASE, id, 415, AvailableOptionEnum, console, log, concat, AVAILABLE, __webpack_module_cache__, __webpack_require__, moduleId, exports, module
<i> [webpack.Progress]  | 151 ms sealing > asset processing
<i> [webpack.Progress] 176 ms sealing
 ERROR  Failed in compile:

HookWebpackError: "." is not in the SourceMap.
    at makeWebpackError (proj/node_modules/webpack/lib/HookWebpackError.js:49:9)
    at proj/node_modules/webpack/lib/Compilation.js:1982:11
    at eval (eval at create (proj/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:34:1)
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:321:16
    at proj/node_modules/neo-async/async.js:2830:7
    at proj/node_modules/neo-async/async.js:2830:7
    at done (proj/node_modules/neo-async/async.js:2863:11)
    at proj/node_modules/neo-async/async.js:2818:7
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:314:19
    at proj/node_modules/webpack/lib/Cache.js:93:5```

**Comment by chaffeqa** [comment]
I also now have a solution for our situation (dont know whose bug this is honestly)...

We need to transpile `enum`s a different way, so I did the following:

Installed and used: https://github.com/dosentmatter/babel-plugin-const-enum (the constObject approach since it will produce a `const obj = {}` instead of that iife)

```
{
  "plugins": [
    [
      "const-enum",
      {
        "transform": "constObject"
      }
    ]
  ]
}```

I then went and globally replaced all our `export enum ...` with `export const enum ...` so they get targeted by the plugin.

**Comment by chaffeqa** [comment]
uuugh ran into another instance that throws the same error:

```ts
async function createDbRecordFor(tableName: string, attrs: any) {
  return null
}

class DataLayerBase {
  public createDbRecordFor = createDbRecordFor
}
export default DataLayerBase
```

and the error:
```
HookWebpackError: "." is not in the SourceMap.
    at makeWebpackError (proj/node_modules/webpack/lib/HookWebpackError.js:49:9)
    at proj/node_modules/webpack/lib/Compilation.js:1982:11
    at eval (eval at create (proj/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:34:1)
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:321:16
    at proj/node_modules/neo-async/async.js:2830:7
    at proj/node_modules/neo-async/async.js:2830:7
    at done (proj/node_modules/neo-async/async.js:2863:11)
    at proj/node_modules/neo-async/async.js:2818:7
    at proj/node_modules/webpack/lib/SourceMapDevToolPlugin.js:314:19
    at proj/node_modules/webpack/lib/Cache.js:93:5
```

source:
```
'/* harmony export */ __webpack_require__.d(__webpack_exports__, {\n' +
    '/* harmony export */   "default": () => __WEBPACK_DEFAULT_EXPORT__\n' +
    '/* harmony export */ });\n' +
    'function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }\n' +
    '\n' +
    'function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }\n' +
    '\n' +
    'function createDbRecordFor(_x, _x2) {\n' +
    '  return _createDbRecordFor.apply(this, arguments);\n' +
    '}\n' +
    '\n' +
    'function _createDbRecordFor() {\n' +
    '  _createDbRecordFor = _asyncToGenerator(function* (tableName, attrs) {\n' +
    '    return null;\n' +
    '  });\n' +
    '  return _createDbRecordFor.apply(this, arguments);\n' +
    '}\n' +
    '\n' +
    'class DataLayerBase {\n' +
    '  constructor() {\n' +
    '    this.createDbRecordFor = createDbRecordFor;\n' +
    '  }\n' +
    '\n' +
    '}\n' +
    '\n' +
    '/* harmony default export */ const __WEBPACK_DEFAULT_EXPORT__ = (DataLayerBase);\n' +
    '\n' +
    '/***/ })\n' +
    '\n' +

```

So i figured out, if i removed the `async` it compiles.

**Comment by chaffeqa** [comment]
both of these scenarios resulting in the same error leads me to believe that instead of it compiling wrong, I think maybe terser is eliminating dead code possibly (or what it thinks is dead)... which produces that ambiguous entry error?

**Comment by evilebottnawi** [comment]
@chaffeqa Please open a new issue

**Comment by evilebottnawi** [comment]
Looks like you have broken source maps, maybe problem on babel side, anyway can you open new issue?

**Comment by chaffeqa** [comment]
yep! would you like the new issue in this repo or somewhere on babel side? (I agree, i think maybe something with babel-typescript?)

**Comment by evilebottnawi** [comment]
@chaffeqa let's open here, also please provide minimum reproducible case, and I investigate it

**Comment by MarcelloPerathoner** [comment]
@sokra This was fixed in @1.4.1 but the fix only partially made it into @2.0.0. 

Line 144 in lib/applySourceMap.js@2.0.0 should be:

```const sourceContent = sourceMapConsumer.sourceContentFor(source, true);```

**Comment by evilebottnawi** [comment]
@MarcelloPerathoner Can you create reproducible test repo?
